apps/examples/wifi_manager/wm_test/wm_test_connectbyrssi_test.c

```diff
@@ -0,0 +1,311 @@
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <semaphore.h>
+#include <errno.h>
+#include <wifi_manager/wifi_manager.h>
+#include <netutils/netlib.h>
+#include "wm_test.h"
+#include "wm_test_log.h"
+
+static sem_t g_wio_sem;
+static struct wt_options *opt;
+static char bssid[20];
+static char connected_ap_mac[20];
+
+#define TAG "[WTCR]"
+
+#define WIO_TEST_SIGNAL								\
+	do {											\
+		sem_post(&g_wio_sem);			\
+		WT_LOG(TAG, "send signal");	\
+	} while (0)
+
+#define WIO_TEST_WAIT								\
+	do {											\
+		WT_LOG(TAG, "wait signal");	\
+		sem_wait(&g_wio_sem);				\
+	} while (0)
+
+
+static void _wt_sta_connected(wifi_manager_cb_msg_s msg, void* arg);
+static void _wt_sta_disconnected(wifi_manager_cb_msg_s msg, void* arg);
+static void _wt_scan_done(wifi_manager_cb_msg_s msg, void* arg);
+static void _wt_print_scanlist(wifi_manager_scan_info_s * list);
+
+static wifi_manager_cb_s g_wifi_callbacks = {
+	_wt_sta_connected,
+	_wt_sta_disconnected,
+	NULL, 
+	NULL,
+	_wt_scan_done,
+};
+
+/* Supported security method */
+static const char *g_wifi_test_auth_method[] = {
+#ifdef WT_AUTH_TABLE
+#undef WT_AUTH_TABLE
+#endif
+#define WT_AUTH_TABLE(type, str, desc) str,
+#include "wm_test_auth_table.h"
+};
+
+static const char *g_wifi_test_crypto_method[] = {
+#ifdef WT_CRYPTO_TABLE
+#undef WT_CRYPTO_TABLE
+#endif
+#define WT_CRYPTO_TABLE(type, str, desc) str,
+#include "wm_test_crypto_table.h"
+};
+
+void _wt_sta_connected(wifi_manager_cb_msg_s msg, void* arg)
+{
+	WT_LOG(TAG, "-->res(%d)", msg.res);		
+	WT_LOG(TAG, "bssid %02x:%02x:%02x:%02x:%02x:%02x",
+		msg.bssid[0], msg.bssid[1],
+		msg.bssid[2], msg.bssid[3],
+		msg.bssid[4], msg.bssid[5]);
+
+	snprintf(connected_ap_mac, sizeof(connected_ap_mac), "%02x:%02x:%02x:%02x:%02x:%02x",
+		msg.bssid[0], msg.bssid[1],
+		msg.bssid[2], msg.bssid[3],
+		msg.bssid[4], msg.bssid[5]);	
+
+	if (WIFI_MANAGER_SUCCESS == msg.res) {
+		WT_LOG(TAG, "Station Connected");
+	} else {
+		WT_LOGE(TAG, "Station not Connected");
+	}
+	WIO_TEST_SIGNAL;
+}
+
+void _wt_sta_disconnected(wifi_manager_cb_msg_s msg, void* arg)
+{	
+	WT_LOG(TAG, "--> res(%d) reason %d", msg.res, msg.reason);
+	WIO_TEST_SIGNAL;
+}
+
+void _wt_scan_done(wifi_manager_cb_msg_s msg, void* arg)
+{
+	WT_LOG(TAG, "-->");
+	/* Make sure you copy the scan results onto a local data structure.
+	 * It will be deleted soon eventually as you exit this function.
+	 */
+	if (msg.res != WIFI_MANAGER_SUCCESS || msg.scanlist == NULL) {
+		WIO_TEST_SIGNAL;
+		return;
+	}
+	_wt_print_scanlist(msg.scanlist);
+	wifi_manager_scan_info_s* wifi_scan_iter = msg.scanlist;
+	while (wifi_scan_iter != NULL) {
+		if (strncmp(wifi_scan_iter->ssid, opt->ssid, WIFIMGR_SSID_LEN) == 0) {
+			memcpy(bssid, wifi_scan_iter->bssid, strlen(wifi_scan_iter->bssid));
+			break;
+		}
+		wifi_scan_iter = wifi_scan_iter->next;
+	}
+	WIO_TEST_SIGNAL;
+}
+
+
```

Comment:
Could you please avoid this double line space

Reply:
Done